spec/models/user_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  let(:user) { build(:user) }
+  let(:user2) { build(:user) }
+  let(:user3) { build(:user) }
+
+  it 'creates valid user' do
+    expect(user.save).to be(true)
+  end
+
+  context 'is not saved because it' do
+    it 'forbides blank username' do
+      user = build(:user, username: '')
+      user.save
+      expect(user.errors[:username]).to include("can't be blank")
+    end
+
+    it 'requires valid email' do
+      user = build(:user, email: '')
+      user.save
+      expect(user.errors[:email]).to include("can't be blank")
+
+      user.email = 'wrongmicro.com'
+      user.save
+      expect(user.errors[:email]).to include('is invalid')
+    end
+
+    it 'requires password to contain at least 6 characters' do
```

Comment:
Ditto, we don't need to write a spec for password as we can trust `devise` to do its job properly in this regard.